src/Cafe/OS/libs/nsyshid/BackendLibusb.cpp

```diff
@@ -694,22 +694,53 @@ namespace nsyshid::backend::libusb
 			return false;
 		}
 
-		// ToDo: implement this
-#if 0
-		// is this correct? Discarding "ifIndex" seems like a bad idea
-		int ret = libusb_set_configuration(handleLock->getHandle(), protocol);
-		if (ret == 0) {
-			cemuLog_logDebug(LogType::Force,
-							 "nsyshid::DeviceLibusb::setProtocol(): success");
-			return true;
+		struct libusb_config_descriptor* conf = nullptr;
+		libusb_device* dev = libusb_get_device(handleLock->GetHandle());
+		int ret = libusb_get_active_config_descriptor(dev, &conf);
```

Comment:
This probably leaks some memory: The [docs](https://libusb.sourceforge.io/api-1.0/group__libusb__desc.html#ga425885149172b53b3975a07629c8dab3) say that `conf` must be freed with `libusb_free_config_descriptor()`.
However, the result of `libusb_get_device` does not need to be freed. The joys of programming in C :)
To cite the docs: "do not feel compelled to unreference it when you are done".

Reply:
good catch, have added that in now!